### PR TITLE
Add dependencies print for Quarkus added artifacts

### PR DESCRIPTION
**Pull request by gtroitsk**
This PR extends current RHBQ artifact diff job with dependencies print of added/new artifacts. This Java program will be used in Jenkins job later.
After running program `mvn clean install exec:java -Dquarkus.maven.dir="path_to_dir" -Dquarkus.new-artifacts-list="added_artifacts.txt"`, the result is saved to file added_artifacts_deps.txt

### CODE REVIEW

File: diff-utils/src/main/java/org/quarkus/qe/AddedArtifactsPrint.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
+package org.quarkus.qe;
```
Comment (mjurc) [inline diff-utils/src/main/java/org/quarkus/qe/AddedArtifactsPrint.java]: Let's go with `io.quarkus.qe` package name, as we do for our framework/test suite/etc.

File: diff-utils/src/main/java/org/quarkus/qe/Artifact.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
+package org.quarkus.qe;
```
Comment (mjurc) [inline diff-utils/src/main/java/org/quarkus/qe/Artifact.java]: Let's go with `io.quarkus.qe` package name, as we do for our framework/test suite/etc.

File: diff-utils/src/main/java/org/quarkus/qe/Coordinates.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package org.quarkus.qe;
```
Comment (mjurc) [inline diff-utils/src/main/java/org/quarkus/qe/Coordinates.java]: Let's go with `io.quarkus.qe` package name, as we do for our framework/test suite/etc.

File: diff-utils/src/main/java/org/quarkus/qe/Main.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package org.quarkus.qe;
```
Comment (mjurc) [inline diff-utils/src/main/java/org/quarkus/qe/Main.java]: Let's go with `io.quarkus.qe` package name, as we do for our framework/test suite/etc.

File: diff-utils/src/main/java/org/quarkus/qe/MavenRepo.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
+package org.quarkus.qe;
```
Comment (mjurc) [inline diff-utils/src/main/java/org/quarkus/qe/MavenRepo.java]: Let's go with `io.quarkus.qe` package name, as we do for our framework/test suite/etc.

File: diff-utils/src/main/java/org/quarkus/qe/VersionedCoordinates.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
+package org.quarkus.qe;
```
Comment (mjurc) [inline diff-utils/src/main/java/org/quarkus/qe/VersionedCoordinates.java]: Let's go with `io.quarkus.qe` package name, as we do for our framework/test suite/etc.

File: diff-utils/pom.xml [hunk 7]
```diff
@@ -0,0 +1,95 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.quarkus.qe</groupId>
```
Comment (mjurc) [inline diff-utils/pom.xml]: Let's go with `io.quarkus.qe` package name, as we do for our framework/test suite/etc.

File: diff-utils/pom.xml [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    <artifactId>added-artifacts</artifactId>
```
Comment (mjurc) [inline diff-utils/pom.xml]: `diff-utils` maybe? Added artifacts seems a little too specific :]

File: README.md [hunk 9]
```diff
@@ -1,8 +1,10 @@
...
 
-##Existing utilities
+## Existing utilities
 
 ### diff-utils
 
-`create_colorful_diff.sh` utility to create  the colorfull diff between two files.
+- `create_colorful_diff.sh` utility to create  the colorful diff between two files.
+- Java program that prints dependencies of added artifacts \
+`mvn clean install exec:java -Dquarkus.maven.dir="path_to_dir" -Dquarkus.new-artifacts-list="added_artifacts.txt"`
```
Comment (mjurc) [inline README.md:10]: This is a good first step, but imho this should be integrated into the diff script later on - the colourful diff after all is where we want to have the dependencies shown.

**Review by mjurc — CHANGES_REQUESTED** [review]
I can't get this to work.

I've tested this with 3.2.6.CR1 - https://main-jenkins-csb-quarkusqe.apps.ocp-c1.prod.psi.redhat.com/view/all/job/rhbq-platform-quarkus-maven-repo-gav-diff/18/artifact/artifacts_3.2.6.CR1_ADDED.txt - this is the added artifact list. The Maven repository zip used can be found in that run's parameters:

```
mvn clean install exec:java -Dquarkus.new-artifacts-list=/home/mjurc/Documents/quarkus-utilities/diff-utils/artifacts_3.2.6.CR1_ADDED.txt -Dquarkus.maven.dir=/home/mjurc/Downloads/rh-quarkus-platform-3.2.6.GA-maven-repository/maven-repository
```

yielded an empty file, so did

```
mvn clean install exec:java -Dquarkus.new-artifacts-list=/home/mjurc/Documents/quarkus-utilities/diff-utils/artifacts_3.2.6.CR1_ADDED.txt -Dquarkus.maven.dir=/home/mjurc/Downloads/rh-quarkus-platform-3.2.6.GA-maven-repository/maven-repository
```

Maybe you could prepare a Jenkins job where it would work, @gtroitsk ? You can take https://main-jenkins-csb-quarkusqe.apps.ocp-c1.prod.psi.redhat.com/view/all/job/rhbq-platform-quarkus-maven-repo-gav-diff/18/ as baseline.

File: diff-utils/pom.xml (hunk 8 above)
Comment (gtroitsk) [reply diff-utils/pom.xml]: Changed

File: README.md (hunk 9 above)
Comment (gtroitsk) [reply README.md:10]: Ok, I will adjust README to use single script as I have implementation for extended color diff.  This PR only prints dependencies of new artifact in simple text format.
Please see https://main-jenkins-csb-quarkusqe.apps.ocp-c1.prod.psi.redhat.com/view/utils/job/utils-quarkus-maven-repo-gav-diff-gtroitsk/20/artifact/artifacts_3.2.6.CR1_ADDED.txt

**Comment by mjurc** [comment]
Hey, I see you've updated the other items - but can you describe what you ran this script with? I cannot figure that step out at all, see https://github.com/quarkus-qe/quarkus-utilities/pull/2#pullrequestreview-1672374252 - can you address that? Thanks!

**Comment by gtroitsk** [comment]
I use the list of added artifacts in format of groupId:artifactId ($GA in Jenkins script) without versions etc and pass it as a property `quarkus.new-artifacts-list` to Java program.
You can see `added_artifacts_list.txt` in the job configuration script: https://main-jenkins-csb-quarkusqe.apps.ocp-c1.prod.psi.redhat.com/view/utils/job/utils-quarkus-maven-repo-gav-diff-gtroitsk/configure

**Comment by mjurc** [comment]
Oh, that report looks correct. I think we can merge this as first step. Thank you!

The next steps:
1. create a script in this repo to run this new Maven project with the outputs of the other script from the existing diff job (or integrate it there)
2. add this dependency info into the colourful diff - make the added dependencies clickable and show/hide what brings them in on click

**Review by mjurc — APPROVED** [review]

**Pull request merged**